File: app/src/main/java/antoninovitale/dropcodechallenge/details/section/model/Status.java
package antoninovitale.dropcodechallenge.details.section.model;

/**
 * Created by antoninovitale on 31/08/2017.
 */
public enum Status {
    IDLE, RUNNING, PAUSED, DONE;

    public static String getStatusLabel(Status s, String idle, String running, String paused,
                                        String done) {
        String label = null;
        switch (s) {
            case IDLE:
                label = idle;
                break;
            case RUNNING:
                label = running;
                break;
            case PAUSED:
                label = paused;
                break;
            case DONE:
                label = done;
                break;
            default:
                break;
        }
        return label;
    }

}

File: app/src/main/java/com/ninovitale/punkapi/app/details/section/model/Status.kt
package com.ninovitale.punkapi.app.details.section.model

/**
 * Created by antoninovitale on 31/08/2017.
 */
enum class Status {
    IDLE, RUNNING, PAUSED, DONE;

    companion object {
        @JvmStatic
        fun getStatusLabel(
                s: Status?,
                idle: String?,
                running: String?,
                paused: String?,
                done: String?
        ): String? {
            var label: String? = null
            when (s) {
                IDLE -> label = idle
                RUNNING -> label = running
                PAUSED -> label = paused
                DONE -> label = done
                else -> {
                }
            }
            return label
        }
    }
}